stdlib/stomp/src/main/ballerina/stomp/sender.bal

```diff
@@ -0,0 +1,174 @@
+import ballerina/socket;
+import ballerina/io;
+import ballerina/log;
+import ballerina/system;
+
+# Configurations related to a STOMP connection
+#
+# + host - STOMP provider url.
+# + port - STOMP port.
+# + config - Config.
+# + login - STOMP user login.
+# + passcode - STOMP passcode.
+# + vhost - default stomp vhost.
+# + acceptVersion - 1.1.
+# + socketClient - socketConnection.
+# + endOfFrame - null octet.
+
+public type Sender client object {
+
+    public string host = "";
+    public int port = 0;
+    public string login = "";
+    public string passcode = "";
+    public string vhost = "";
+    public string acceptVersion = "";
+    private socket:Client socketClient;
+
+    // End of frame used a null octet (^@ = \u0000).
+    public string endOfFrame = "\u0000";
+
+    public ConnectionConfiguration config = {
+        host:host,
+        port:port,
+        login:login,
+        passcode:passcode,
+        vhost:vhost,
+        acceptVersion:acceptVersion
+    };
+
+    public function __init(ConnectionConfiguration stompConfig) returns error? {
+        self.config = stompConfig;
+        self.host = stompConfig.host;
+        self.port = stompConfig.port;
+        self.login = stompConfig.login;
+        self.passcode = stompConfig.passcode;
+        self.vhost = stompConfig.vhost;
+        self.acceptVersion = stompConfig.acceptVersion;
+        self.socketClient = new({
+                host: self.host,
+                port: self.port,
+                callbackService: ClientService
+            });
+        var connection = self->connect(stompConfig);
+    }
+
+    public remote function connect(ConnectionConfiguration stompConfig) returns error?;
+
+    public remote function send(string message, string destination) returns error?;
+
+    public remote function disconnect() returns error?;
+};
+
+public type ConnectionConfiguration record {
+    string host;
+    int port;
+    string login;
+    string passcode;
+    string vhost;
+    string acceptVersion;
+};
+
+public remote function Sender.connect(ConnectionConfiguration stompConfig) returns error?{
+    socket:Client socketClient = self.socketClient;
+    io:println("Starting up the Ballerina Stomp Service\n");
+
+    // CONNECT frame to get connected.
+    string connect = "CONNECT" + "\n" +
+        "accept-version:" + stompConfig.acceptVersion + "\n" +
+        "login:" + stompConfig.login + "\n" +
+        "passcode:" + stompConfig.passcode + "\n" +
+        "host:" + stompConfig.vhost + "\n" +
+        "\n" + self.endOfFrame;
+
+    byte[] payloadByte = connect.toByteArray("UTF-8");
+    // Send desired content to the server using the write function.
+    var writeResult = socketClient->write(payloadByte);
+    if (writeResult is error) {
+        io:println("Unable to write the connect frame", writeResult);
+    }
```

Comment:
We shouldn't be handling errors here. Errors should be propagated to the user in all places